test/integration/smoke/test_router_dnsservice.py

```diff
@@ -0,0 +1,268 @@
+        cls.services["virtual_machine"]["displayname"] = 'drump';
+        cls.services["virtual_machine"]["name"] = 'drump';
+        cls.vm1 = VirtualMachine.create(cls.api_client,
+                                         cls.services["virtual_machine"],
+                                         templateid=cls.template.id,
+                                         accountid=cls.account.name,
+                                         domainid=cls.domain.id,
+                                         serviceofferingid=cls.service_offering.id,
+                                         networkids=[str(cls.network.id)])
+        cls.vm1.password = "password"
+        cls.logger.debug("Created VM named drump");
+                
+        cls.services["virtual_machine"]["displayname"] = 'dilton';
+        cls.services["virtual_machine"]["name"] = 'dilton';
+        cls.vm2 = VirtualMachine.create(cls.api_client,
+                                         cls.services["virtual_machine"],
+                                         templateid=cls.template.id,
+                                         accountid=cls.account.name,
+                                         domainid=cls.domain.id,
+                                         serviceofferingid=cls.service_offering.id,
+                                         networkids=[str(cls.network.id)])
+        cls.vm2.password = "password"
+        cls.logger.debug("Created VM named dilton");
+
+        cls.services["natrule1"] = {
+            "privateport": 22,
+            "publicport": 22,
+            "protocol": "TCP"
+        }
+
+        cls.services["configurableData"] = {
+            "host": {
+                "password": "password",
+                "username": "root",
+                "port": 22
+            },
+            "input": "INPUT",
+            "forward": "FORWARD"
+        }
+
+        cls._cleanup = [
+            cls.vm1,
+            cls.vm2,
+            cls.network,
+            cls.network_offering,
+            cls.service_offering,
+            cls.account
+        ]
+
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.api_client, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.cleanup = []
+
+
+    def tearDown(self):
+        try:
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+
+    def test_router_common(self):
+        """Performs common router tests and returns router public_ips"""
+
+        routers = list_routers(
+            self.apiclient,
+            account=self.account.name,
+            domainid=self.account.domainid
+        )
```

Comment:
This is followed by a set of assertions to ensure that that there is a router with valid state.